src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -208,7 +207,7 @@ def construct_manifest_parameters(self) -> Dict[str, Any]:
                 "publisherName": {"value": self.config.publisher_name},
                 "acrArtifactStoreName": {"value": self.config.acr_artifact_store_name},
                 "acrManifestName": {"value": self.config.acr_manifest_name},
-                "armTemplateName": {"value": self.config.arm_template.artifact_name},
+                "armTemplateName": {"value": f"{self.config.network_function_definition_group_name}_nfd_artifact"},
```

Comment:
It might be better if this name is a property in the _configuration.py file. You defined it twice (once here and once in nsd_generator.py. If someone changes one but not the other, it might lead to issues